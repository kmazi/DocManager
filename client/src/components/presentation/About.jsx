import React from 'react';

/**
 * Renders the about page of the application
 * @return {object} - Returns the html to render
 */
const About = () => (
  <div className="aboutpage container">
    <h3 className="center-align">About DocManger</h3>
    <hr />
    <p>
      DocManger is an electronic document management system
      that helps you manage your
      documents for easy sharing and accessing.
      It offers features such as creating, editing,
      restricting and sharing personal documents.
    </p>
  </div>);


export default About;
