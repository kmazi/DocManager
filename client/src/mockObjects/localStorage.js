/**
 * Mocks localstorage
 */
class LocalStorageMock {
  /**
   * the constructor
   */
  constructor() {
    this.store = {};
  }
/**
 * clears the localstorage
 * @return {void} returns null
 */
  clear() {
    this.store = {};
  }

/**
 * get item in the localstorage
 * @param {string} key - The key to the item in the localstorage
 * @return {string} returns the key to the item in localstorage
 */
  getItem(key) {
    return this.store[key];
  }
/**
 * set the value of an item in the localstorage class
 * @param {string} key - The key to the item in the localstorage
 * @param {string} value - The value of the item in the localstorage
 * @return {null} returns null
 */
  setItem(key, value) {
    this.store[key] = value;
  }

  /**
 * remove item in the localstorage class
 * @param {string} key - The key to the item in the localstorage
 * @return {void} returns nothing
 */
  removeItem(key) {
    delete this.store[key];
  }
}

global.localStorage = new LocalStorageMock();
export default global.localStorage;
