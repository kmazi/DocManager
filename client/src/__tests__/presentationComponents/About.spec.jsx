import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import About from '../../components/presentation/About';
import '../../mockObjects/localStorage';

describe('The About component:', () => {
  const props = {
  };
  test('should render correctly', () => {
    const enzymeWrapper = mount(<About {...props} />);
    const component = enzymeWrapper;
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
