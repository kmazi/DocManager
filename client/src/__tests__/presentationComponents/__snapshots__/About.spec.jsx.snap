// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The About component: should render correctly 1`] = `
<About>
  <div
    className="aboutpage container"
  >
    <h3
      className="center-align"
    >
      About DocManger
    </h3>
    <hr />
    <p>
      DocManger is an electronic document management system that helps you manage your documents for easy sharing and accessing. It offers features such as creating, editing, restricting and sharing personal documents.
    </p>
  </div>
</About>
`;
